test/test_parsy.py

```diff
@@ -365,6 +366,78 @@ def test_decimal_digit(self):
                          "9876543210")
         self.assertRaises(ParseError, decimal_digit.parse, "¹")
 
+    def test_line_info(self):
+        @generate
+        def foo():
+            i = yield line_info
+            l = yield any_char
+            return (l, i)
+
+        self.assertEqual(foo.many().parse("AB\nCD"),
+                         [("A", (0, 0)), ("B", (0, 1)),
+                          ("\n", (0, 2)),
+                          ("C", (1, 0)), ("D", (1, 1)),
+                          ])
+
+
+class TestParserTokens(unittest.TestCase):
+    """
+    Tests that ensure that `.parse` can handle an arbitrary list of tokens,
+    rather than a string.
+    """
+    # Some opaque objects we will use in our stream:
```

Comment:
I usually just use `START = object()` for this purpose; this has the advantage of being explicit about what it does (create a new unique object).

Reply:
I used a class as it would get me a `repr` that includes the name, while `object()` does not, so that parse failures at least show something useful. But then, it looks like I never use that fact because the `repr` is ugly...